### PR TITLE
bpo-29569: threading.Timer class: Continue periodical execution till action return True

### PR DESCRIPTION
**Pull request by slytomcat**
I think that functionality of threading.Timer class can be easily extended to generate the sequence of runs with specified period. The idea comes from the GLib.timeout_add function. 
 
http://bugs.python.org/issue29569

As most current CB functions that are used in Timer returns nothing (None) they will run only once as earlier. Only functions that returns True will continue their periodical execution. 

There are two ways to stop such continues execution:
- Return something that is not True from action function,
- Call cancel method of Timer class.   

It is my first contribution and I kindly ask to help me with required future actions.

Thanks

### CODE REVIEW

**Comment by slytomcat** [comment]
I've fixed test for continuous execution of Timer class. Now check passed.

**Comment by andrewnester** [comment]
I am wondering how it will influence existing code. Most of the functions currently used in Timer callback probably not returning True value. It means that they will be executed infinitely.

Maybe it's better to add some new argument like `repeat_until` which takes some callback indicating when to finish execution. By default it will have some value to indicate only 1 repeat of callback execution.

**Comment by slytomcat** [comment]
No, exactly conversely - if function returns None or not True it will run only once.
The only returning True continues execution after next timeout.

**Comment by andrewnester** [comment]
@slytomcat right, sorry, my bad

**Comment by slytomcat** [comment]
codecov/patch — 81.25% of diff hit (target 100%) - anybody know how to fix it?

File: Lib/threading.py [hunk 1]
```diff
@@ -1177,9 +1177,10 @@ def cancel(self):
         self.finished.set()
 
     def run(self):
-        self.finished.wait(self.interval)
-        if not self.finished.is_set():
-            self.function(*self.args, **self.kwargs)
+        """Continue execution after wait until function returns True"""
+        while(not self.finished.wait(self.interval)):
```
Comment (andrewnester) [inline Lib/threading.py]: Try to add test case when this is true. It might help you get 100% coverage

Comment (slytomcat) [reply Lib/threading.py]: Done. But: codecov/patch — 86.95% of diff hit (target 100%) - some red lines in  Lib/test/test_threading.py. 
I have no idea how to cover tests by tests =/

**Comment by slytomcat** [comment]
I have to reorganize my cloned repo so the CR have to be reissued.

**Comment by slytomcat** [comment]
Reopened CR: https://github.com/python/cpython/pull/191

**Pull request closed without merging**